ADDING_NEW_RULES.md

```diff
@@ -0,0 +1,94 @@
+
+All classes that implements rules output are under the [`notice` package](https://github.com/MobilityData/gtfs-validator/tree/master/domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice) of the domain layer
+
+They must extend either `ErrorNotice` or `WarningNotice` and fit the `*Notice` format
+
+Start by adding the error code definition to the corresponding `ErrorNotice` or `WarningNotice` class located in the `base` package, for example
+
+`protected static final String E_0XX = "E0XX";`
+
+You can see here in the constructor signature what fields your own Notice class will inherit
+
+~~~
+public ErrorNotice(String filename,
+                   String noticeId,
+                   String title,
+                   String description,
+                   String entityId)
+~~~
+
+Add your new class in the `error` or `warning` package accordingly
+
+You can define fields and getters that are not already covered by the base classes
+
+Your new class must override and define an `export` method like the following
+
+    @Override
+    public void export(final NoticeExporter exporter) throws IOException {
+        exporter.export(this);
+    }
+
+### 3. Implement your new rule exportation code
+
+Add a new definition at the bottom of the [`NoticeExporter` interface](https://github.com/MobilityData/gtfs-validator/blob/master/domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/NoticeExporter.java)
+
+Add the required code to the interface implementations in the [`exporter` package](https://github.com/MobilityData/gtfs-validator/tree/master/adapter/exporter/src/main/java/org/mobilitydata/gtfsvalidator/exporter)
+
+**Note**: JSON export required at a minimum. Leaving the protobuf implementation empty will not lead to your PR being rejected.
+
+### 4. Implement exportation code unit tests
+
+Add a new test implementation to the [`*ExporterTest` classes](https://github.com/MobilityData/gtfs-validator/tree/master/adapter/exporter/src/test/java/org/mobilitydata/gtfsvalidator/exporter)
+
+### 5. Implement your new rule in a specific use case
+
+All rules in the validator are implemented in self-contained classes in the [`usecase` package](https://github.com/MobilityData/gtfs-validator/tree/master/usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase)
+
+Add a class constructor which should take two finalized parameters
+
+- a `GtfsDataRepository` from which the feed content can be pulled
+- a `ValidationResultRepository` to which any warning or error encountered can be pushed
+
+Implement the verification code in the `execute` method pushing instances of your new `Notice` class upon detecting issues
+
+### 6. Implement use case unit tests
+
+Add a Test class for your new use case in the [following location](https://github.com/MobilityData/gtfs-validator/tree/master/usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase)
+
+### 7. Add a way to instantiate the new use case
```

Comment:
I'd combine steps 7 and 8 into one - I think it's clearer, as they are effectively the last step of actually running the validation rule.